### PR TITLE
Add a checks provider filter for catalog_new view

### PR DESCRIPTION
**Pull request by EMaksy**
# Description
Add a filter option for the catalog_new view so incoming checks from wanda can be filtered like priviously in the catalog view

[Screencast from 2023-01-24 12-22-19.webm](https://user-images.githubusercontent.com/54111255/214279234-384192aa-35a6-4f6a-a29c-be79ae6e3939.webm)

## How was this tested?

Added test for the ProviderLabels component functions.
Reworking the existing ChecksCatalogNew test

### CODE REVIEW

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx [hunk 1]
```diff
@@ -1,66 +1,81 @@
...
   const dispatch = useDispatch();
-
+  const [selected, setSelected] = useState(providers[0]);
+  const provider = selected;
   const {
     data: catalogData,
     error: catalogError,
     loading,
-  } = useSelector(getCatalog());
+  } = useSelector(getCatalog(provider));
```
Comment (arbulu89) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:24]: The `getCatalog` selector doesn't accept any incoming param.
In any case, you don't need it, as this only gets the current redux state.
Simply revert the change

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx [hunk 2]
```diff
@@ -1,66 +1,81 @@
...
-            <div
-              key={idx}
-              className="check-group bg-white shadow overflow-hidden sm:rounded-md mb-8"
-            >
-              <div className="bg-white px-4 py-5 border-b border-gray-200 sm:px-6">
-                <h3 className="text-lg leading-6 font-medium text-gray-900">
-                  {group}
-                </h3>
+    <div>
+      <ProviderSelection
```
Comment (arbulu89) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:40]: mmm, in not sure if this `ProviderSelection` shouldn't go inside the `CatalogContainer`, to don't show the filter if the API to wanda fails

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx [hunk 3]
```diff
@@ -1,66 +1,81 @@
...
 import CatalogContainer from './CatalogContainer';
 import CheckItem from './CheckItem';
+import ProviderSelection from './ProviderSelection';
 
+const providers = ['default'].concat(providersList.slice());
 // eslint-disable-next-line import/prefer-default-export
 export function ChecksCatalogNew() {
   const dispatch = useDispatch();
-
+  const [selected, setSelected] = useState(providers[0]);
```
Comment (arbulu89) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:18]: Call this directly `provider` and rename maybe `setSelected` by `setSelectedProvider`

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx [hunk 4]
```diff
@@ -1,66 +1,81 @@
...
 import { groupBy } from '@lib/lists';
 
 import { getCatalog } from '@state/selectors/catalog';
 import { updateCatalog } from '@state/actions/catalog';
+import { providersList } from '../ProviderLabel/ProviderLabel';
 import CatalogContainer from './CatalogContainer';
 import CheckItem from './CheckItem';
+import ProviderSelection from './ProviderSelection';
 
+const providers = ['default'].concat(providersList.slice());
```
Comment (arbulu89) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:14]: I think the labels approach is wrong, as you would be sending the labels in the query, and we might not recognize them. We would need to send the keys:
We don't want to send `provider=AWS` we want to send `provider=aws`

File: assets/js/components/ProviderLabel/ProviderLabel.jsx [hunk 5]
```diff
@@ -50,4 +50,14 @@ function ProviderLabel({ provider }) {
   );
 }
 
+const getLabels = (labelValue) => {
```
Comment (arbulu89) [inline assets/js/components/ProviderLabel/ProviderLabel.jsx:53]: You could have this as:
```
const getLabels = () =>
    Object.values(providerData).map((value) => value.label);
```

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx [hunk 6]
```diff
@@ -1,66 +1,81 @@
...
-    dispatch(updateCatalog());
-  }, [dispatch]);
+    setSelected(providers[0]);
+  }, [providers[0]]);
 
   useEffect(() => {
-    getCatalog();
-  }, []);
+    if (provider === 'default') {
+      dispatch(updateCatalog());
```
Comment (arbulu89) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:32]: I don't think we need this distinction.
If you set provider to `null`, it makes the same work

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx (hunk 4 above)
Comment (arbulu89) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:14]: I wouldn't call this `default`. I will name `All`, because we would get all the checks

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hey @EMaksy ,
I think we need to revisit some things, as the code uses the labels to create the queries, and we need to use the provider key.

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx [hunk 7]
```diff
@@ -1,17 +1,30 @@
...
 import CatalogContainer from './CatalogContainer';
 import CheckItem from './CheckItem';
+import ProviderSelection from './ProviderSelection';
+
+const ALL_FILTER = 'All';
 
 // eslint-disable-next-line import/prefer-default-export
 export function ChecksCatalogNew() {
   const dispatch = useDispatch();
+  const updatedProvider = providerData;
```
Comment (arbulu89) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:23]: Hey @EMaksy ,
I have looked at this, and we cannot do this direct assignment, as this simply copies the reference of the variable, so changing `updatedProvider` changes `providerData` afterwards. In order to avoid this we need to do a shallow copy, like this: `const updatedProviderData = {...providerData, '': ALL_FILTER}`.
This shallow copies the variable and adds the new empty field to the new object.

After that, we need to change where `providerData` is used by `updatedProviderData` all around the code in this file

Comment (EMaksy) [reply assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx]: Hey good point, i applied the  suggested changes. Thanks

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx [hunk 8]
```diff
@@ -1,17 +1,31 @@
...
 
 import { groupBy } from '@lib/lists';
 
 import { getCatalog } from '@state/selectors/catalog';
 import { updateCatalog } from '@state/actions/catalog';
+import {
+  providerData,
+  getLabels,
+  getProviderByLabel,
+} from '../ProviderLabel/ProviderLabel';
```
Comment (dottorblaster) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:13]:
```suggestion
} from '@components/ProviderLabel/ProviderLabel';
```

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Last touch!

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx [hunk 9]
```diff
@@ -20,47 +34,61 @@ export function ChecksCatalogNew() {
...
-                    key={check.id}
-                    checkID={check.id}
-                    premium={check.premium}
-                    description={check.description}
-                    remediation={check.remediation}
-                  />
-                ))}
-              </ul>
-            </div>
+    <div>
```
Comment (arbulu89) [inline assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:45]: I think we can safely remove this `div`

**Review by arbulu89 — APPROVED** [review]
Perfect! Thanks @EMaksy

File: assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx (hunk 9 above)
Comment (EMaksy) [reply assets/js/components/ChecksCatalog/ChecksCatalogNew.jsx:45]: yes :D

**Review by dottorblaster — APPROVED** [review]
LGTM
